### ISSUE TITLE
Планы на выпуск от 2022.02.30

### ISSUE THREAD

**Issue by ventureoo** [ventureoo/ARU#59]
Используйте эту тему для предложения новых изменений которые могут войти в следующий выпуск. Текущий же план таков:
- [ ] Переработать раздел "Разгон монитора" (См. https://github.com/ventureoo/ARU/issues/58)
- [ ] Переработать раздел "Кастомные ядра"
  - [ ] Переименовать раздел из "Кастомные ядра для увелечения плавности и зачем это нужно" просто в "Кастомные ядра".
  - [ ] Добавить новые ядра  
  - [ ] Улучшить описание различных настроек
  - [ ] Дать развернутый ответ на вопрос "Зачем нужны кастомные ядра?" 
- [ ] Добавить раздел с описанием использования технологии DLSS под Proton (https://github.com/ventureoo/ARU/pull/55).
- [ ] Улучшить индексацию HTML страниц.

На этом пока все. Пишите свои предложения ниже.

**Comment by dewdpol** [comment]
Думаю для dlss нужно дождаться релиза proton-ge-custom на стабильном wine

**Comment by dewdpol** [comment]
С кастомными ядрами интересный момент... я провёл , пока что, поверхностное сравнение стокового ядра с xanmod и у меня в ashes if singularity получился результат не в пользу кастомного, но результат очень поверхностный и без выборки

**Comment by dewdpol** [comment]
@ventureoo в кастомные ядра возможно стоит добавить информацию на счет планировщиков и как их выбирать при сборке, если я правильно помню, то также думаю стоит добавить информацию о kernel timer frequency.

**Comment by ventureoo** [comment]
@dewdpol Да, по планировщикам много информации придется перерабатывать. Во-первых, мы долгое время в ARU рекомендовали использовать MuQSS, ибо по нашим тестам он давал лучшие задержки при достаточно хорошем FPS в играх. Но к сожалению [его поддержка и разработка прекращена](https://www.opennet.ru/opennews/art.shtml?num=55737). Последнее патчи что я находил были на версию 5.14. Но понятно что никто тянуть лямку дальше не станет. uPDS похоже тоже окончательно умер. Не смотря на то, что его вроде как тянул мейнтейнер linux-tkg, последняя версия в которой его можно применить это 5.11. Из более менее "мейнстримных" планировщиков в живых остались только PDS и BMQ. Оба предоставляются на выбор в linux-lqx, хотя по умолчанию отдается предпочтение PDS. Первый мы с @Almarus когда-то тестировали и он нам в свое время не понравился, но судя по моим последним замерам он уже стал выдавать хорошие результаты и его можно использовать, впрочем таких низких задержек как в MuQSS я получить так и не смог. BMQ я до сих пор никак не тестировал, поэтому ничего про него сказать не могу. А вот новых планровщиков появилось вагон и маленькая тележка: https://github.com/hamadmarri/cacule-cpu-scheduler, https://github.com/hamadmarri/TT-CPU-Scheduler, https://github.com/hamadmarri/Baby-CPU-Scheduler (все три от одного разработчика кстати), https://github.com/firelzrd/bore-scheduler. Их и надо затестить.

**Comment by ventureoo** [comment]
Попрошу всех теперь отказаться от использования 

> :bash:\`code block\`

для встраивания части кода прямо в текст. Вместо этого используйте два обратных апострофа до и после текста который должен находится в блоке кода. Например:

> \`\`code block\`\`

@dewdpol 

См. https://github.com/ventureoo/ARU/commit/7e42d238e6ffc4c70d45070d676b78f4530dae61

**Comment by dewdpol** [comment]
@ventureoo нужна помощь в переформатировании?

**Comment by ventureoo** [comment]
> @ventureoo нужна помощь в переформатировании?

Нет, спасибо.

**Comment by dewdpol** [comment]
@ventureoo на счёт ядер: необходимо добавить информацию о таких ядрах как: linux-ck, linux-clear, linux-mptcp, linux-pf, linux-rt, linux-vfio?
Ядра типа linux-libre, linux-aufs думаю, что смысла не имеют, поскольку один лишен проприетарных драйверов, а второй заявлен для docker. Также не знаю на счёт linux-mainline, linux-next-gut и linux-git, думаю эти тоже спорные.

Если перечисленные ядра те, что нужны могу попробовать заняться для начала оформлением базовой информации, а потом взависимости от доступного времени

**Comment by ventureoo** [comment]
@dewdpol Проблема всех ядер указанными вами выше в том, что они как правило улучшают что-то одно, а нам нужно чтобы улучшения были в совокупности и работали хорошо.

`linux-ck`

Поддержка патчей автором брошена и обновляться они не будут. Но их поддержку взял на себя автор Xanmod. То есть, все патчи CK сейчас есть в Xanmod.
https://github.com/xanmod/linux-patches/tree/master/linux-5.15.y-xanmod/ck-hrtimer
Будь там обновленный планировщик MuQSS я бы подумал, но так не вижу смысла.

`linux-clear`

Все патчи clearlinux уже давно растасканы по другим ядрам. Они есть и в lqx, и в Xanmod, в linux-tkg, и даже, если мне память не изменяет, есть и в Zen.

`linux-mptcp`

Я не слышал о таком ядре, но судя по тому что я увидел - он базируется на ветке ядра 4.19 и содержит только сетевые улучшения. Не вижу смысла его добавлять.

`linux-pf`

Достойное упоминания ядро, только вот многие что там уже есть тоже давно есть и в других ядрах. Если тестирования покажут что он имеет какое-то преимущество - лишним не будет.

`linux-rt` 

Realtime патч есть в [linux-xanmod-rt](https://aur.archlinux.org/packages/linux-xanmod-rt) и недавно появился в [linux-tkg](https://github.com/Frogging-Family/linux-tkg/pull/380) (Тут стоит только упомянуть опцию `_preempt_rt`)

`linux-vfio` 

Аналогично с linux-clear. ACS патч есть в [linux-zen](https://archlinux.org/packages/?name=linux-zen), linux-lqx, linux-xanmod, linux-tkg (Нужно включить параметр [_acs_override](https://github.com/Frogging-Family/linux-tkg/blob/master/customization.cfg#L148))

**Comment by ventureoo** [comment]
Лично я сейчас заинтересован ядром [linux-cachy](https://github.com/CachyOS/linux-cachyos), ибо он имеет расширенный выбор планировщика, да и само ядро по моим ощущениям показывает себя очень хорошо.

**Comment by ventureoo** [comment]
@Atmosphelen Я не против, у меня самого возникала идея сделать раздел про браузеры чтобы описать включение аппаратного ускорения декодирования видео. Думаю такая информация больше подходит для first-steps и выбора ПО.

**Comment by ventureoo** [comment]
В основном нужно смотреть на потребление памяти и скорость загрузки страниц. С последним правда сложно делать замеры, возможно есть какие-то бенчмарки которые позволяют это делать.

**Comment by dewdpol** [comment]
@ventureoo отсылка композитора от полноэкранных окон? Если я правильно понимаю, то выключить композитинг у полноэкранных окон можно и без kwin-lowlatency, если об этом речь, насколько помню возможно даже в автоматическом режиме, но нужно проверить

**Comment by ventureoo** [comment]
@dewdpol Нет, KWin по умолчанию не делает отвязки композитора от полноэкранных окон, по крайне мере для X11 сессий. У него есть опция блокировки композитинга другими приложениями, но это совершенно не то. Только для Wayland начиная с версии 5.22 у него есть такя возможность.

**Comment by KuKuRuZ1337** [comment]
> `linux-rt`
> 
> Realtime патч есть в [linux-xanmod-rt](https://aur.archlinux.org/packages/linux-xanmod-rt) и недавно появился в [linux-tkg](https://github.com/Frogging-Family/linux-tkg/pull/380) (Тут стоит только упомянуть опцию `_preempt_rt`)

Смысл-то от rt, если из-за него в играх FPS меньше ? 
А на сколько я помню: "Чем выше ФПС -- тем меньше задержки".
А от бесмысленного RT ФПС всегда меньше и из этого следует, что больше задержки
P.S. у меня монитор 165герц и я не ощущаю никаких задержик, но более меньший ФПС уже даёт по глазам

**Comment by ventureoo** [comment]
@KuKuRuZ1337 ARU не нацелен только на оптимизацию под игрушки. RT патч на декстопах используется людьми которые хотят свести задержки звука к минимуму.

**Comment by ventureoo** [comment]
Я немного удивлен, но похоже что дефолтный планировщик CFS с последними версиями ядра довольно сильно улучшили, да настолько что для меня он теперь ЛУЧШЕ чем PDS. Хотелось бы сделать тесты на этот счёт, да и возможно это просто так работают твики liqorix (zen) ядра.

**Comment by dewdpol** [comment]
@ventureoo я читал, что относительно недавно дефолтный планировщик улучшали, то ли в 5,15, то ли чуть раньше.

**Comment by Organarh** [comment]
Сколько должен устанавливаться wine-tkg-git ? У меня уже 1 ч 20 м. собираетсяэ
-fno-strict-aliasing -Wdeclaration-after-statement -Wempty-body -Wignored-qualifiers -Winit-self \
  -Wno-packed-not-aligned -Wshift-overflow=2 -Wstrict-prototypes -Wtype-limits \
  -Wunused-but-set-parameter -Wvla -Wwrite-strings -Wpointer-arith -Wlogical-op -Wabsolute-value \
  -Wno-format -Wformat-overflow -Wnonnull -mcx16 -Wformat -O2 -ftree-vectorize -march=native
ccache x86_64-w64-mingw32-gcc -c -o dlls/wbemprox/qualifier.cross.o ../wine-mirror-git/dlls/wbemprox/qualifier.c -Idlls/wbemprox \
  -I../wine-mirror-git/dlls/wbemprox -Iinclude -I../wine-mirror-git/include \
  -I../wine-mirror-git/include/msvcrt -D__WINESRC__ -D_UCRT -D__WINE_PE_BUILD -Wall \
  -fno-strict-aliasing -Wdeclaration-after-statement -Wempty-body -Wignored-qualifiers -Winit-self \
  -Wno-packed-not-aligned -Wshift-overflow=2 -Wstrict-prototypes -Wtype-limits \
  -Wunused-but-set-parameter -Wvla -Wwrite-strings -Wpointer-arith -Wlogical-op -Wabsolute-value \
  -Wno-format -Wformat-overflow -Wnonnull -mcx16 -Wformat -O2 -ftree-vectorize -march=native

**Comment by ventureoo** [comment]
@Organarh В зависимости от процессора. Как правило часа полтора-два.
